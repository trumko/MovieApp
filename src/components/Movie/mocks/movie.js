export const movie = {
  "id": 338970,
  "title": "Tomb Raider",
  "tagline": "Her legend begins",
  "vote_average": 6.2,
  "vote_count": 817,
  "release_date": "2018-03-08",
  "poster_path": "https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
  "overview": "Lara Croft, the fiercely independent daughter of a missing adventurer, must push herself beyond her limits when she finds herself on the island where her father disappeared.",
  "budget": 94000000,
  "revenue": 126025000,
  "genres": [
    "Action",
    "Adventure"
  ],
  "runtime": 118
}
