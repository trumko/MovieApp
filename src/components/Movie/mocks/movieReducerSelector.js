export const movieReducerSelector = {
  movieReducer: {
    movie: {
      id: 449176,
      title: 'Love, Simon',
      tagline: 'He\'s done keeping his story straight.',
      vote_average: 8.4,
      vote_count: 82,
      release_date: '2018-03-16',
      poster_path: 'https://image.tmdb.org/t/p/w500/5YUYg5q7QfC4IoNwNUtiwdiYKPr.jpg',
      overview: 'Everyone deserves a great love story. But for seventeen-year old Simon Spier it’s a little more complicated: he’s yet to tell his family or friends he’s gay and he doesn’t actually know the identity of the anonymous classmate he’s fallen for online.',
      budget: 17000000,
      revenue: 11800000,
      genres: [
        'Comedy',
        'Romance',
        'Drama'
      ],
      runtime: 110
    },
    similarResults: [
      {
        id: 19404,
        title: 'Dilwale Dulhania Le Jayenge',
        tagline: 'Come... Fall in Love',
        vote_average: 9.2,
        vote_count: 1297,
        release_date: '1995-10-20',
        poster_path: 'https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg',
        overview: 'Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancé. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.',
        budget: 13200000,
        revenue: 100000000,
        genres: [
          'Comedy',
          'Drama',
          'Romance'
        ],
        runtime: 190
      },
      {
        id: 418827,
        title: 'Sister\'s Younger Husband',
        tagline: 'A sexy brother-in-law approached her? And he’s a younger man?',
        vote_average: 8.5,
        vote_count: 2,
        release_date: '2016-10-11',
        poster_path: 'https://image.tmdb.org/t/p/w500/wKCkyb7dyDTbZn7qy6zh8oqtFrT.jpg',
        overview: 'A sexy brother-in-law approached her? And he’s a younger man?  Mi-yeon, who was adopted into her family when she was little, is living with her older sister, Si-yeon and Si-yeon’s husband. Mi-yeon’s desire towards her brother-in-law grows more and more every night when she gets to watch the affectionate interaction between her older sister and the brother-in-law. One day, Mi-yeon seduces her brother-in-law and the two end up crossing the line when they should not have. As the situation goes on, Mi-yeon finds out her older sister is also having affairs outside the marriage. Mi-yeon tries to clear up the relationship with her brother-in-law out of guilty feeling. However, she hears something shocking from the brother-in-law.',
        budget: 0,
        revenue: 0,
        genres: [
          'Comedy',
          'Drama',
          'Romance'
        ],
        runtime: 75
      },
      {
        id: 449176,
        title: 'Love, Simon',
        tagline: 'He\'s done keeping his story straight.',
        vote_average: 8.4,
        vote_count: 82,
        release_date: '2018-03-16',
        poster_path: 'https://image.tmdb.org/t/p/w500/5YUYg5q7QfC4IoNwNUtiwdiYKPr.jpg',
        overview: 'Everyone deserves a great love story. But for seventeen-year old Simon Spier it’s a little more complicated: he’s yet to tell his family or friends he’s gay and he doesn’t actually know the identity of the anonymous classmate he’s fallen for online.',
        budget: 17000000,
        revenue: 11800000,
        genres: [
          'Comedy',
          'Drama',
          'Romance'
        ],
        runtime: 110
      },
      {
        id: 455661,
        title: 'In a Heartbeat',
        tagline: 'The Heart Wants What The Heart Wants',
        vote_average: 8.3,
        vote_count: 370,
        release_date: '2017-06-01',
        poster_path: 'https://image.tmdb.org/t/p/w500/wJUJROdLOtOzMixkjkx1aaZGSLl.jpg',
        overview: 'A closeted boy runs the risk of being outed by his own heart after it pops out of his chest to chase down the boy of his dreams.',
        budget: 0,
        revenue: 0,
        genres: [
          'Family',
          'Animation',
          'Romance',
          'Comedy'
        ],
        runtime: 4
      },
      {
        id: 637,
        title: 'Life Is Beautiful',
        tagline: 'An unforgettable fable that proves love, family and imagination conquer all.',
        vote_average: 8.3,
        vote_count: 4899,
        release_date: '1997-12-20',
        poster_path: 'https://image.tmdb.org/t/p/w500/f7DImXDebOs148U4uPjI61iDvaK.jpg',
        overview: 'A touching story of an Italian book seller of Jewish ancestry who lives in his own little fairy tale. His creative and happy life would come to an abrupt halt when his entire family is deported to a concentration camp during World War II. While locked up he tries to convince his son that the whole thing is just a game.',
        budget: 20000000,
        revenue: 229400000,
        genres: [
          'Comedy',
          'Drama'
        ],
        runtime: 116
      },
      {
        id: 13,
        title: 'Forrest Gump',
        tagline: 'Life is like a box of chocolates...you never know what you\'re gonna get.',
        vote_average: 8.3,
        vote_count: 10297,
        release_date: '1994-07-06',
        poster_path: 'https://image.tmdb.org/t/p/w500/yE5d3BUhE8hCnkMUJOo1QDoOGNz.jpg',
        overview: 'A man with a low IQ has accomplished great things in his life and been present during significant historic events - in each case, far exceeding what anyone imagined he could do. Yet, despite all the things he has attained, his one true love eludes him. \'Forrest Gump\' is the story of a man who rose above his challenges, and who proved that determination, courage, and love are more important than ability.',
        budget: 55000000,
        revenue: 677945399,
        genres: [
          'Comedy',
          'Drama',
          'Romance'
        ],
        runtime: 142
      },
      {
        id: 901,
        title: 'City Lights',
        tagline: 'True Blind Love',
        vote_average: 8.2,
        vote_count: 581,
        release_date: '1931-01-30',
        poster_path: 'https://image.tmdb.org/t/p/w500/bXNvzjULc9jrOVhGfjcc64uKZmZ.jpg',
        overview: 'City Lights is the first silent film that Charlie Chaplin directed after he established himself with sound accompanied films. The film is about a penniless man who falls in love with a flower girl. The film was a great success and today is deemed a cult classic.',
        budget: 1500000,
        revenue: 19181,
        genres: [
          'Comedy',
          'Drama',
          'Romance'
        ],
        runtime: 87
      },
      {
        id: 77338,
        title: 'The Intouchables',
        tagline: 'Sometimes you have to reach into someone else\'s world to find out what\'s missing in your own.',
        vote_average: 8.2,
        vote_count: 6940,
        release_date: '2011-11-02',
        poster_path: 'https://image.tmdb.org/t/p/w500/4mFsNQwbD0F237Tx7gAPotd0nbJ.jpg',
        overview: 'A true story of two men who should never have met – a quadriplegic aristocrat who was injured in a paragliding accident and a young man from the projects.',
        budget: 13000000,
        revenue: 426480871,
        genres: [
          'Drama',
          'Comedy'
        ],
        runtime: 112
      },
      {
        id: 914,
        title: 'The Great Dictator',
        tagline: 'Once again - the whole world laughs!',
        vote_average: 8.2,
        vote_count: 991,
        release_date: '1940-10-15',
        poster_path: 'https://image.tmdb.org/t/p/w500/hK2KsU0lNiYRqIrYJcC3x63dPnH.jpg',
        overview: 'Dictator Adenoid Hynkel tries to expand his empire while a poor Jewish barber tries to avoid persecution from Hynkel\'s regime.',
        budget: 2000000,
        revenue: 11000000,
        genres: [
          'Comedy'
        ],
        runtime: 125
      },
      {
        id: 3082,
        title: 'Modern Times',
        tagline: 'He stands alone as the greatest entertainer of modern times! No one on earth can make you laugh as heartily or touch your heart as deeply...the whole world laughs, cries and thrills to his priceless genius!',
        vote_average: 8.2,
        vote_count: 1174,
        release_date: '1936-02-05',
        poster_path: 'https://image.tmdb.org/t/p/w500/7uoiKOEjxBBW0AgDGQWrlfGQ90w.jpg',
        overview: 'The Tramp struggles to live in modern industrial society with the help of a young homeless woman.',
        budget: 1,
        revenue: 8500000,
        genres: [
          'Drama',
          'Comedy'
        ],
        runtime: 87
      },
      {
        id: 239,
        title: 'Some Like It Hot',
        tagline: 'The movie too HOT for words!',
        vote_average: 8.1,
        vote_count: 1069,
        release_date: '1959-03-18',
        poster_path: 'https://image.tmdb.org/t/p/w500/pxc9EFCMYkItESpqqrI783yl8Gh.jpg',
        overview: 'Two musicians witness a mob hit and struggle to find a way out of the city before they are found by the gangsters. Their only opportunity is to join an all-girl band as they leave on a tour. To make their getaway they must first disguise themselves as women, then keep their identities secret and deal with the problems this brings - such as an attractive bandmate and a very determined suitor.',
        budget: 2883848,
        revenue: 25000000,
        genres: [
          'Comedy',
          'Romance'
        ],
        runtime: 122
      },
      {
        id: 105,
        title: 'Back to the Future',
        tagline: 'He\'s the only kid ever to get into trouble before he was born.',
        vote_average: 8.1,
        vote_count: 7789,
        release_date: '1985-07-03',
        poster_path: 'https://image.tmdb.org/t/p/w500/pTpxQB1N0waaSc3OSn0e9oc8kx9.jpg',
        overview: 'Eighties teenager Marty McFly is accidentally sent back in time to 1955, inadvertently disrupting his parents\' first meeting and attracting his mother\'s romantic interest. Marty must repair the damage to history by rekindling his parents\' romance and - with the help of his eccentric inventor friend Doc Brown - return to 1985.',
        budget: 19000000,
        revenue: 381109762,
        genres: [
          'Adventure',
          'Comedy',
          'Science Fiction',
          'Family'
        ],
        runtime: 116
      }
    ]
  }
};
