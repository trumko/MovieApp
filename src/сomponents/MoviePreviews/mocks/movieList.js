import inBrugesPoster from 'assets/in_bruges_poster.jpg';
import lostInTranslationPoster from 'assets/lost_in_translation_poster.jpg';

export const movieList = [
  {
    title: 'In Bruges',
    date: '2009',
    genre: 'drama, comedie',
    posterUrl: inBrugesPoster,
    id: 1,
  }, {
    title: 'Lost in Transl...',
    date: '2003',
    genre: 'drama',
    posterUrl: lostInTranslationPoster,
    id: 2,
  }, {
    title: 'In Bruges',
    date: '2008',
    genre: 'drama, comedy',
    posterUrl: inBrugesPoster,
    id: 3,
  }, {
    title: 'Lost in Transl...',
    date: '2003',
    genre: 'drama',
    posterUrl: lostInTranslationPoster,
    id: 4,
  },  {
    title: 'In Bruges',
    date: '2009',
    genre: 'drama, comedy',
    posterUrl: inBrugesPoster,
    id: 5,
  },  {
    title: 'Lost in Transl...',
    date: '2003',
    genre: 'drama',
    posterUrl: lostInTranslationPoster,
    id: 6,
  },
];
