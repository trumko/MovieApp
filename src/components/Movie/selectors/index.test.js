import { movieReducerSelector } from '../mocks/movieReducerSelector'
import { selectMovie, selectSimilarResults } from './index'

describe('Movie Selectors', () => {
  test('should return movie', () => {
    const selected = selectMovie(movieReducerSelector);
    expect(selected).toMatchSnapshot();
  });

  test('should return similar movies', () => {
    const selected = selectSimilarResults(movieReducerSelector);
    expect(selected).toMatchSnapshot();
  });
});
