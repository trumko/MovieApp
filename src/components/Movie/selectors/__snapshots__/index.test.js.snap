// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Selectors should return movie 1`] = `
Object {
  "budget": 17000000,
  "genres": Array [
    "Comedy",
    "Romance",
    "Drama",
  ],
  "id": 449176,
  "overview": "Everyone deserves a great love story. But for seventeen-year old Simon Spier it’s a little more complicated: he’s yet to tell his family or friends he’s gay and he doesn’t actually know the identity of the anonymous classmate he’s fallen for online.",
  "poster_path": "https://image.tmdb.org/t/p/w500/5YUYg5q7QfC4IoNwNUtiwdiYKPr.jpg",
  "release_date": "2018-03-16",
  "revenue": 11800000,
  "runtime": 110,
  "tagline": "He's done keeping his story straight.",
  "title": "Love, Simon",
  "vote_average": 8.4,
  "vote_count": 82,
}
`;

exports[`Movie Selectors should return similar movies 1`] = `
Array [
  Object {
    "budget": 13200000,
    "genres": Array [
      "Comedy",
      "Drama",
      "Romance",
    ],
    "id": 19404,
    "overview": "Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancé. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.",
    "poster_path": "https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
    "release_date": "1995-10-20",
    "revenue": 100000000,
    "runtime": 190,
    "tagline": "Come... Fall in Love",
    "title": "Dilwale Dulhania Le Jayenge",
    "vote_average": 9.2,
    "vote_count": 1297,
  },
  Object {
    "budget": 0,
    "genres": Array [
      "Comedy",
      "Drama",
      "Romance",
    ],
    "id": 418827,
    "overview": "A sexy brother-in-law approached her? And he’s a younger man?  Mi-yeon, who was adopted into her family when she was little, is living with her older sister, Si-yeon and Si-yeon’s husband. Mi-yeon’s desire towards her brother-in-law grows more and more every night when she gets to watch the affectionate interaction between her older sister and the brother-in-law. One day, Mi-yeon seduces her brother-in-law and the two end up crossing the line when they should not have. As the situation goes on, Mi-yeon finds out her older sister is also having affairs outside the marriage. Mi-yeon tries to clear up the relationship with her brother-in-law out of guilty feeling. However, she hears something shocking from the brother-in-law.",
    "poster_path": "https://image.tmdb.org/t/p/w500/wKCkyb7dyDTbZn7qy6zh8oqtFrT.jpg",
    "release_date": "2016-10-11",
    "revenue": 0,
    "runtime": 75,
    "tagline": "A sexy brother-in-law approached her? And he’s a younger man?",
    "title": "Sister's Younger Husband",
    "vote_average": 8.5,
    "vote_count": 2,
  },
  Object {
    "budget": 17000000,
    "genres": Array [
      "Comedy",
      "Drama",
      "Romance",
    ],
    "id": 449176,
    "overview": "Everyone deserves a great love story. But for seventeen-year old Simon Spier it’s a little more complicated: he’s yet to tell his family or friends he’s gay and he doesn’t actually know the identity of the anonymous classmate he’s fallen for online.",
    "poster_path": "https://image.tmdb.org/t/p/w500/5YUYg5q7QfC4IoNwNUtiwdiYKPr.jpg",
    "release_date": "2018-03-16",
    "revenue": 11800000,
    "runtime": 110,
    "tagline": "He's done keeping his story straight.",
    "title": "Love, Simon",
    "vote_average": 8.4,
    "vote_count": 82,
  },
  Object {
    "budget": 0,
    "genres": Array [
      "Family",
      "Animation",
      "Romance",
      "Comedy",
    ],
    "id": 455661,
    "overview": "A closeted boy runs the risk of being outed by his own heart after it pops out of his chest to chase down the boy of his dreams.",
    "poster_path": "https://image.tmdb.org/t/p/w500/wJUJROdLOtOzMixkjkx1aaZGSLl.jpg",
    "release_date": "2017-06-01",
    "revenue": 0,
    "runtime": 4,
    "tagline": "The Heart Wants What The Heart Wants",
    "title": "In a Heartbeat",
    "vote_average": 8.3,
    "vote_count": 370,
  },
  Object {
    "budget": 20000000,
    "genres": Array [
      "Comedy",
      "Drama",
    ],
    "id": 637,
    "overview": "A touching story of an Italian book seller of Jewish ancestry who lives in his own little fairy tale. His creative and happy life would come to an abrupt halt when his entire family is deported to a concentration camp during World War II. While locked up he tries to convince his son that the whole thing is just a game.",
    "poster_path": "https://image.tmdb.org/t/p/w500/f7DImXDebOs148U4uPjI61iDvaK.jpg",
    "release_date": "1997-12-20",
    "revenue": 229400000,
    "runtime": 116,
    "tagline": "An unforgettable fable that proves love, family and imagination conquer all.",
    "title": "Life Is Beautiful",
    "vote_average": 8.3,
    "vote_count": 4899,
  },
  Object {
    "budget": 55000000,
    "genres": Array [
      "Comedy",
      "Drama",
      "Romance",
    ],
    "id": 13,
    "overview": "A man with a low IQ has accomplished great things in his life and been present during significant historic events - in each case, far exceeding what anyone imagined he could do. Yet, despite all the things he has attained, his one true love eludes him. 'Forrest Gump' is the story of a man who rose above his challenges, and who proved that determination, courage, and love are more important than ability.",
    "poster_path": "https://image.tmdb.org/t/p/w500/yE5d3BUhE8hCnkMUJOo1QDoOGNz.jpg",
    "release_date": "1994-07-06",
    "revenue": 677945399,
    "runtime": 142,
    "tagline": "Life is like a box of chocolates...you never know what you're gonna get.",
    "title": "Forrest Gump",
    "vote_average": 8.3,
    "vote_count": 10297,
  },
  Object {
    "budget": 1500000,
    "genres": Array [
      "Comedy",
      "Drama",
      "Romance",
    ],
    "id": 901,
    "overview": "City Lights is the first silent film that Charlie Chaplin directed after he established himself with sound accompanied films. The film is about a penniless man who falls in love with a flower girl. The film was a great success and today is deemed a cult classic.",
    "poster_path": "https://image.tmdb.org/t/p/w500/bXNvzjULc9jrOVhGfjcc64uKZmZ.jpg",
    "release_date": "1931-01-30",
    "revenue": 19181,
    "runtime": 87,
    "tagline": "True Blind Love",
    "title": "City Lights",
    "vote_average": 8.2,
    "vote_count": 581,
  },
  Object {
    "budget": 13000000,
    "genres": Array [
      "Drama",
      "Comedy",
    ],
    "id": 77338,
    "overview": "A true story of two men who should never have met – a quadriplegic aristocrat who was injured in a paragliding accident and a young man from the projects.",
    "poster_path": "https://image.tmdb.org/t/p/w500/4mFsNQwbD0F237Tx7gAPotd0nbJ.jpg",
    "release_date": "2011-11-02",
    "revenue": 426480871,
    "runtime": 112,
    "tagline": "Sometimes you have to reach into someone else's world to find out what's missing in your own.",
    "title": "The Intouchables",
    "vote_average": 8.2,
    "vote_count": 6940,
  },
  Object {
    "budget": 2000000,
    "genres": Array [
      "Comedy",
    ],
    "id": 914,
    "overview": "Dictator Adenoid Hynkel tries to expand his empire while a poor Jewish barber tries to avoid persecution from Hynkel's regime.",
    "poster_path": "https://image.tmdb.org/t/p/w500/hK2KsU0lNiYRqIrYJcC3x63dPnH.jpg",
    "release_date": "1940-10-15",
    "revenue": 11000000,
    "runtime": 125,
    "tagline": "Once again - the whole world laughs!",
    "title": "The Great Dictator",
    "vote_average": 8.2,
    "vote_count": 991,
  },
  Object {
    "budget": 1,
    "genres": Array [
      "Drama",
      "Comedy",
    ],
    "id": 3082,
    "overview": "The Tramp struggles to live in modern industrial society with the help of a young homeless woman.",
    "poster_path": "https://image.tmdb.org/t/p/w500/7uoiKOEjxBBW0AgDGQWrlfGQ90w.jpg",
    "release_date": "1936-02-05",
    "revenue": 8500000,
    "runtime": 87,
    "tagline": "He stands alone as the greatest entertainer of modern times! No one on earth can make you laugh as heartily or touch your heart as deeply...the whole world laughs, cries and thrills to his priceless genius!",
    "title": "Modern Times",
    "vote_average": 8.2,
    "vote_count": 1174,
  },
  Object {
    "budget": 2883848,
    "genres": Array [
      "Comedy",
      "Romance",
    ],
    "id": 239,
    "overview": "Two musicians witness a mob hit and struggle to find a way out of the city before they are found by the gangsters. Their only opportunity is to join an all-girl band as they leave on a tour. To make their getaway they must first disguise themselves as women, then keep their identities secret and deal with the problems this brings - such as an attractive bandmate and a very determined suitor.",
    "poster_path": "https://image.tmdb.org/t/p/w500/pxc9EFCMYkItESpqqrI783yl8Gh.jpg",
    "release_date": "1959-03-18",
    "revenue": 25000000,
    "runtime": 122,
    "tagline": "The movie too HOT for words!",
    "title": "Some Like It Hot",
    "vote_average": 8.1,
    "vote_count": 1069,
  },
  Object {
    "budget": 19000000,
    "genres": Array [
      "Adventure",
      "Comedy",
      "Science Fiction",
      "Family",
    ],
    "id": 105,
    "overview": "Eighties teenager Marty McFly is accidentally sent back in time to 1955, inadvertently disrupting his parents' first meeting and attracting his mother's romantic interest. Marty must repair the damage to history by rekindling his parents' romance and - with the help of his eccentric inventor friend Doc Brown - return to 1985.",
    "poster_path": "https://image.tmdb.org/t/p/w500/pTpxQB1N0waaSc3OSn0e9oc8kx9.jpg",
    "release_date": "1985-07-03",
    "revenue": 381109762,
    "runtime": 116,
    "tagline": "He's the only kid ever to get into trouble before he was born.",
    "title": "Back to the Future",
    "vote_average": 8.1,
    "vote_count": 7789,
  },
]
`;
